### PR TITLE
Types Declaration (Modified)

### PR DESCRIPTION
**Pull request by climaxmba**
# Add Types Declaration: #26
I have removed the dependency in **Pull #51**. It is not required for the declaration to work.

### CODE REVIEW

**Comment by climaxmba** [comment]
> I have removed the dependency in **Pull #51** and `tsconfig.json`. They are not required for the declaration to work.

**Comment by climaxmba** [comment]
I'm sorry about #51 trying to import `typescipt`. I didn't realize the incorrect spelling.

File: package-lock.json
Comment (Naramsim) [inline package-lock.json:1]: Hi, can you remove this file? You are still changing all our dependencies.

File: @types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    getPokemonColorByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokemonFormByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokemonHabitatByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokemonShapeByName(name: string | number | string[] | number[]): Promise<object>;
+    getPokemonSpeciesByName(name: string | number | string[] | number[]): Promise<object>;
+    getStatByName(name: string | number | string[] | number[]): Promise<object>;
+    getTypeByName(name: string | number | string[] | number[]): Promise<object>;
+    getLanguageByName(name: string | number | string[] | number[]): Promise<object>;
+
+    getEndpointsList(): Promise<object>;
```
Comment (Naramsim) [inline @types/index.d.ts:64]: All these methods support passing a `dict` optional parameter:

https://github.com/PokeAPI/pokeapi-js-wrapper#root-endpoints-list

```json
{
  offset: 34,
  limit: 10,
}
```

Is it possible to document it?

File: @types/index.d.ts
Comment (Naramsim) [inline @types/index.d.ts:1]: Could you move this file to the root? Getting rid of the folder `@types`?

File: package.json
Comment (Naramsim) [inline package.json:1]: Could you add the file to the `.files[]` property?

**Review by Naramsim — CHANGES_REQUESTED** [review]

File: @types/index.d.ts (hunk 1 above)
Comment (climaxmba) [reply @types/index.d.ts:64]: Sure, but how about the `resource` method? Does it also have the `dict` parameter?

Comment (climaxmba) [reply @types/index.d.ts]: Alright, It's all done.

**Review by Naramsim — APPROVED** [review]

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    }): Promise<object>;
+    getPokemonFormByName(name: string | number | string[] | number[], dict?: {
+      offset: number,
+      limit: number,
+    }): Promise<object>;
+    getPokemonHabitatByName(name: string | number | string[] | number[], dict?: {
+      offset: number,
+      limit: number,
+    }): Promise<object>;
+    getPokemonShapeByName(name: string | number | string[] | number[], dict?: {
```
Comment (Naramsim) [inline index.d.ts:190]: Hi, sorry, all these methods (`get*byName`, `get*byId`) don't support the options dict.

Comment (climaxmba) [reply index.d.ts]: No problem, done.

**Review by Naramsim — APPROVED** [review]

**Comment by Naramsim** [comment]
Thanks for your time!

**Comment by climaxmba** [comment]
My pleasure, I appreciate your time as well.

**Pull request merged**